freud/box.pyx

```diff
@@ -592,6 +592,12 @@ cdef class Box:
 
         return np.array(l_contains_mask).astype(bool)
 
+    @property
+    def cubic(self):
+        """bool: Whether the box is a cube."""
+        return self.Lx == self.Ly and self.Ly == self.Lz and self.xy == self.yz\
+            and self.yz == self.xz and self.xz == 0
```

Comment:
You may want to use something like np.isclose instead of direct equality in case the boxes are coming from gsd or OVITO and have numerical imprecision introduced somewhere in the file reading/writing.

Reply:
yes, I meant to write myself a comment about this. Thanks for the reminder.

Reply:
A few details to watch for:
- `allclose` is asymmetric, which has consequences:
  - 0 should be the first argument for the tilt factor check to ensure the `rtol` is applied to the potentially nonzero value (tilt factors)
  - it's probably best to check multiple directions for the box lengths, like ensuring that `Lx ~= Ly` and `Ly ~= Lx`. I used an array below to check both directions.
- floats are single precision so NumPy needs a looser `atol`, like `1e-5` instead of `1e-5` for the typical boxes we handle (order of 10^1 length units on each side)

```suggestion
        return np.allclose([self.Lx, self.Lx, self.Ly, self.Ly, self.Lz, self.Lz], [self.Ly, self.Lz, self.Lx, self.Lz, self.Lx, self.Ly], rtol=1e-5, atol=1e-5) and np.allclose(0, [self.xy, self.yz, self.xz], rtol=1e-5, atol=1e-5)
```